src/main/java/duke/command/TaskOperation.java

```diff
@@ -0,0 +1,164 @@
+
+        int taskIndex = Integer.parseInt(userInput.substring(i + 1));
+
+        if(taskIndex > tasks.size()){
+            throw new DukeOutOfRangeException("Please enter a valid index of task :-(");
+        }
+
+        tasks.get(taskIndex - 1).markAsDone();
+        System.out.println(Duke.MESSAGE_BOUNDARY);
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println(tasks.get(taskIndex - 1));
+        System.out.println(Duke.MESSAGE_BOUNDARY);
+
+    }
+
+    public void addTask(String command, String userInput) throws DukeException {
+        int i = userInput.indexOf(" ");
+        String taskInfo = " ";
+        if(i != -1){
+            taskInfo = userInput.substring(i + 1);
+        }
+
+        switch (command) {
+            case "todo":
+                tasks.add(ToDos.parse(taskInfo));
+                break;
+            case "deadline":
+                tasks.add(Deadline.parse(taskInfo));
+                break;
+            case "event":
+                tasks.add(Events.parse(taskInfo));
+                break;
+        }
+
+        System.out.println(Duke.MESSAGE_BOUNDARY + "\n" + "Got it. I've added this task: ");
+        System.out.println(tasks.get(tasks.size() - 1));
+        System.out.println("Now you have " + tasks.size() + " task(s) in the list.\n" + Duke.MESSAGE_BOUNDARY);
+    }
+
+    public void deleteTask(String userInput) throws DukeException{
+        int i = userInput.indexOf(" ");
+        if(i == -1){
+            throw new DukeEmptyParaException("There should be an index of task to delete :-(");
+        }
+
+        int deleteIndex = Integer.parseInt(userInput.substring(i + 1));
+
+        if(deleteIndex > tasks.size()){
+            throw new DukeOutOfRangeException("Please enter a valid index of task :-(");
+        }
+
+        System.out.println(Duke.MESSAGE_BOUNDARY);
+        System.out.println("Noted. I've removed this task:\n" + tasks.get(deleteIndex - 1));
+        tasks.remove(deleteIndex - 1);
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+        System.out.println(Duke.MESSAGE_BOUNDARY);
+
+    }
+
+    public void loadFile(String filePath) throws FileNotFoundException {
+        try{
+            File f = new File(filePath); // create a File for the given file path
+            Scanner s = new Scanner(f); // create a Scanner using the File as the source
+            while(s.hasNext()){
+                loadTask(s.nextLine());
+            }
+            System.out.println("Loading..." + System.lineSeparator() + "███████████████ 100%");
+            printList();
+        } catch (FileNotFoundException e){
+            System.out.println("OOPS!!! No previous record; File not found :-(");
+        }
+    }
+
+    public void loadTask(String s){
+        String[] taskComponent = s.split(" \\| ");
+        char taskType = taskComponent[0].charAt(0);
+        int isDone = Integer.parseInt(taskComponent[1].trim());
```

Comment:
As mentioned before, try not to nest complex logic like this - separate it out into different lines for better code readability